etcdserver/apply.go

```diff
@@ -788,3 +788,19 @@ func isGteRange(rangeEnd []byte) bool {
 func noSideEffect(r *pb.InternalRaftRequest) bool {
 	return r.Range != nil || r.AuthUserGet != nil || r.AuthRoleGet != nil
 }
+
+func removeNeedlessRangeReqs(txn *pb.TxnRequest) {
+	f := func(ops []*pb.RequestOp) {
+		for i := 0; i < len(ops); {
+			switch ops[i].Request.(type) {
+			case *pb.RequestOp_RequestRange:
+				ops = append(ops[:i], ops[i+1:]...)
```

Comment:
probably just nil it out instead of appending? then we can avoid some allocations...